src/compute/cast/mod.rs

```diff
@@ -715,10 +772,156 @@ mod tests {
         assert!((9.0 - c.value(4)).abs() < f64::EPSILON);
     }
 
+    #[test]
+    fn u16_as_u8_overflow() {
+        let array = UInt16Array::from_slice(&[255, 256, 257, 258, 259]);
+        let b = cast(
+            &array,
+            &DataType::UInt8,
+            CastOptions {
+                ignore_overflow: true,
+            },
+        )
+        .unwrap();
+        let c = b.as_any().downcast_ref::<UInt8Array>().unwrap();
+        let values = c.values().as_slice();
+        assert_eq!(values, &[255, 0, 1, 2, 3])
```

Comment:
awesome! What are your thoughts about calling this `wrapping`? This seems the same behavior as `255u8.wrapping_add(1)` vs `255u8.checked_add(1)`

Reply:
Hmm, I did not anything about `wrapping_add ` before.

Reply:
```
pub const fn wrapping_add(self, rhs: Self) -> Self {
            intrinsics::wrapping_add(self, rhs)
        }
```

Seems the arg rhs is `Self`, `255u8.wrapping_add(1u16)` will not work.

Reply:
We have the vectorized versions in `compute/arithmetics` (e.g. `arrow2::compute::arithmetics::basic::add::saturating_add`, since some engines use it to avoid panics (e.g. default behavior of `numpy`). 

Rust `overflowing` is usually used for an operation that returns `Option<T>`, where `None` is when the operation overflows.

Reply:
OK, I got that. But seems lots of duplicate codes should be added.

BTW, would it be possible to introduce a version of `sum`which takes care of the overflow problems? After #56  is done.

```
sum(&array) -> Scalar
```

If the array is U8Array~U64Array, the scalar will be u64;
If the array is I8Array~I64Array, the scalar will be i64;
If the array is floatArray, the scalar will be float64.

In datafuse we have to cast `U8Array` to `U64Array`, then apply the sum with simd.

Reply:
Yes, we can surely do that. Not sure whether we will hit SIMD so easily because there is a cast now, but it is certainly useful. New issue to discuss design?